src/app/clusters/unit-localization-server/unit-localization-server.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "unit-localization-server.h"
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::UnitLocalization;
+using namespace chip::app::Clusters::UnitLocalization::Attributes;
+
+UnitLocalizationServer UnitLocalizationServer::mInstance;
+
+UnitLocalizationServer & UnitLocalizationServer::Instance()
+{
+    return UnitLocalizationServer::mInstance;
+}
+
+CHIP_ERROR UnitLocalizationServer::ReadSupportedTemperatureUnits(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    return aEncoder.Encode(UnitLocalizationServer::Instance().GetSupportedTemperatureUnits());
```

Comment:
Why not:
```c++
aEncoder.Encode(GetSupportedTemperatureUnits());
```
?

However general question might be whether we need these private functions at all? It's a vary simple logic here, so I would just inline them in the `Read()`. And if you would like to keep it, why the signature is `(EndpointId, AttributeValueEncoder &)`, endpoint is not used anywhere here.

The same question for `ReadClusterRevision`.

Reply:
I've addressed your main concern but can't do much changes at the moment because I'm on my smartphone. I'll address this in a follow up PR.